design/one-pager-pluggable-secret-stores.md

```diff
@@ -231,13 +212,20 @@ func (ss *SecretStore) WriteKeyValues(ctx context.Context, s *store.Secret, _ ..
 ```
 
 4. The plugin receives the `ApplySecret` request:
-   1. It will fetch the referenced configuration object, e.g. `VaultConfig`,
-      from the Kubernetes API.
-   2. Using this configuration, it will connect to the External Secret Store,
-      i.e. Vault, and write the secret data.
+    1. It will fetch the referenced configuration object, e.g. `VaultConfig`,
+       from the Kubernetes API.
+    2. Using this configuration, it will connect to the External Secret Store,
+       i.e. Vault, and write the secret data.
 
 ## Alternatives Considered
 
+### Deploy Plugin as a Sidecar
+
+Instead of deploying plugins as separate deployments, it was initially proposed to
+deploy them as sidecars. The main motivation behind this was eliminating authentication
+complications and having a secure communication via sockets. However, adding sidecars
+to Crossplane and providers will make the packaging and deployment more complicated.
```

Comment:
Would you mind elaborating on this a bit? For example:
- We would need to inject plugin sidecars into the core crossplane and to all providers.
- When a new plugin installed at runtime, we would need to inject it to Core Crossplane probably requiring a helm upgrade which couldn't be handled by the package manager unless we build a solution like mutating webhook which diverges from our existing package deployment model.
- When multiple plugins are installed, we would need to have multiple sidecars on multiple nodes. For example, with 3 providers running, if someone want to enable ess plugin for 2 secret stores, we would end up `(3+1) * 2 = 8` plugin processes running as sidecars.